numpy/core/tests/test_multiarray.py

```diff
@@ -5499,6 +5499,28 @@ def test_mean_axis_error(self):
         with assert_raises(np.core._exceptions.AxisError):
             np.arange(10).mean(axis=2)
 
+    def test_mean_where(self):
+        a = np.arange(16).reshape((4, 4))
+        wh_full = np.array([[False, True, False, True],
+                            [True, False, True, False],
+                            [True, True, False, False],
+                            [False, False, True, True]])
+        wh_partial = np.array([[False],
+                               [True],
+                               [True],
+                               [False]])
+        assert_allclose(a.mean(axis=1, where=True),
```

Comment:
A test of `where=False` for mean and var would be good too.

Reply:
Alright, I've added `where=False` tests for `mean`, `var` and `std`, all returning `nan` and raising a `RuntimeWarning`.